### PR TITLE
CI: run tests on `i686` and `wasm32-wasi`

### PR DESCRIPTION
**Pull request by pitdicker**
Fixes https://github.com/chronotope/chrono/issues/352.

### CODE REVIEW

File: src/datetime/tests.rs [hunk 1]
```diff
@@ -1186,7 +1186,10 @@ fn test_subsecond_part() {
 }
 
 #[test]
-#[cfg(feature = "std")]
+#[cfg(all(feature = "std", not(target = "wasm32-wasi")))]
+// Some targets, such as `wasm-wasi` have a problematic definition of `SystemTime`, such as an
+// `i32` (year 2035 problem), or an `u64` (no values before `UNIX-EPOCH`).
+// See https://github.com/rust-lang/rust/issues/44394.
```
Comment (djc) [inline src/datetime/tests.rs:1192]: Nit: I'd like the comments to go before the attributes, keeping the attributes right next to the code.

**Review by djc — APPROVED** [review]

**Comment by djc** [comment]
Does this also fix #1236?

**Comment by pitdicker** [comment]
I have no clue why codecov thinks our coverage in `local/unix/timezone.rs` has massively changed from 85.69% to 66.67%, while at the same time saying the change in that file is -0.14%.

For now codecov seems mostly trouble, but maybe we get some use out of it eventually...

**Review by djc — APPROVED** [review]

**Comment by djc** [comment]
LGTM, thanks!

**Pull request merged**